### ISSUE TITLE
[Request]: Require Type Definitions First

### ISSUE THREAD

**Issue by bradennapier** [gajus/eslint-plugin-flowtype#285]
I can see it being a good rule to enforce all type definitions being at the top of the file. 

**good**
```js
/* @flow */

type MyType = number

const b: MyType = 2

function foo(): MyType {
     return b
}
```

**bad**

```js
/* @flow */

function foo(): MyType {
     return b
}

type MyType = number 

const b: MyType = 2
```

**Comment by pnevyk** [comment]
Related to #168
